soco/soap.py

```diff
@@ -0,0 +1,263 @@
+#   <s:Header>
+#       </Header elements go here>
+#   </s:Header>
+#   <s:Body>
+#       <ns:MethodName xmlns:ns="MethodNamespace>"
+#           <param1>value</param1>
+#           ...
+#           <param_n>value</param_n>
+#       </ns:MethodName>
+#   </s:Body>
+# </s:Envelope>
+
+# pylint: disable=too-many-instance-attributes, too-many-arguments
+# pylint: disable=no-self-use, fixme
+
+
+class SoapMessage(object):
+
+    """ A SOAP Message representing a remote procedure call.
+
+    Uses the Requests library for communication with a SOAP server."""
+
+    def __init__(self, endpoint, method, parameters=None, http_headers=None,
+                 soap_action=None, soap_header=None, namespace=None,
+                 **request_args):
+        """ Args:
+                endpoint (str): The SOAP endpoint URL for this client
+                method (str): The name of the method to call
+                parameters (list): A list of (name, value) tuples containing
+                    the parameters to pass to the method. Default None
+                http_headers (dict): A dict in the form {'Header': 'Value,..}
+                    containing http headers to use for the http request.
+                    Content-type and SOAPACTION headers will be created
+                    automatically, so do not include them here. Use this, for
+                    example, to set a user-agent.
+                soap_action (str): The value of the SOAPACTION header.
+                    Default NONE
+                soap_header (str): A string representation of the XML to be
+                    used for the SOAP Header. Default None
+                namespace (str): The namespace URI to use for the method and
+                    parameters. None, by default.
+                **request_args: Other keyword parameters will be passed to the
+                    Requests request which is used to handle the http
+                    communication. For example, a timeout value can be set.
+        """
+        self.endpoint = endpoint
+        self.method = method
+        self.parameters = [] if parameters is None else parameters
+        self.http_headers = http_headers
+        self.soap_action = soap_action
+        self.soap_header = soap_header
+        self.namespace = namespace
+        self.request_args = request_args
+
+    def prepare_headers(self, http_headers, soap_action):
+        """ Prepare the http headers for sending """
+
+        headers = {'Content-Type': 'text/xml; charset="utf-8"'}
+        if soap_action is not None:
+            headers.update({'SOAPACTION': '"{0}"'.format(soap_action)})
+        if http_headers is not None:
+            headers.update(http_headers)
+        return headers
+
+    def prepare_soap_header(self, soap_header):
+        """ Prepare the SOAP header for sending """
+
+        if soap_header is not None:
+            return '<s:Header>{0}</s:Header>'.format(soap_header)
```

Comment:
It's not (easily) possible here since the `soap_header` parameter is a string representation of XML, not an Element itself.  We could use an Element for `soap_header` by changing the rest of the code, but I think it is probably not worth it on balance (at least as long as this works!!)